test/extended/networking.sh

```diff
@@ -7,9 +7,19 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-if [[ "${OPENSHIFT_QUIET_OUTPUT:-false}" != "true" ]]; then
+CONFIG_ROOT="${OPENSHIFT_CONFIG_ROOT:-}"
+
+# CI mode (deploying clusters for each plugin) should be noisy by
+# default to aid in debugging.
+if [[ "${CONFIG_ROOT}" = "" ]]; then
+  DEFAULT_QUIET=false
+else
+  DEFAULT_QUIET=true
+fi
+
+if [[ "${OPENSHIFT_QUIET_OUTPUT:-${DEFAULT_QUIET}}" != "true" ]]; then
```

Comment:
> Why should the the following test indicate that verbose output was called for? 

Simply because in Bash, the presence of an environment variable is equivalent to a boolean `true` -- if I set `$OPENSHIFT_QUIET_OUTPUT`, I want the output to be quiet.

I understand your gripe about `-n` and `-z`, but they become second-nature fairly quickly. I think the best reason to favor idiomatic Bash is that it's idiomatic. When I write a script that requires a environment variable to equal the string `"true"`, someone else using it will see a boolean environment variable and assume that it's used idiomatically. I recently ran into this when I added jUnit functionality into `hack/test-go.sh`. I personally always pass in `$VAR=true` just to be extra clear (but never use `$VAR=false`), but Clayton used `$VAR=1` when he consumed my script, as that's idiomatic. The script was robust to both approaches as it used the idiomatic Bash expression.

While I agree that Bash is a wily beast, I think our *best* weapon against it is using an idiomatic and consistent approach to writing it. 

If you do `$ man true` or `$ man false` you will also see that `true` and `false` are *not* built-ins, but *programs*. This means that if someone is attempting to do something similar to what you were doing but accidentally evaluated the variable or evaluated `true` when doing the assignment, they'd see incorrect behavior. 

Furthermore, `[`, `[[`, and `if` have an inverse evaluation of numbers-as-booleans, that is `0` is true and `1` (technically, any non-zero number) is false. This is idiomatic for checking the return codes of commands, but not for storing booleans as integers like in C. Even furthermore, the `((` construct evaluates booleans as integers correctly, that is `0` is false and `1` is true. All this being said, trying to represent booleans in Bash using non-Bash-isms is usually harder to read and reason about (which is why I made the comments originally, I had to write a truth table in order to actually understand what your code was doing). 

When I was writing the `os::cmd` suite, I thought a lot about boolean handling in Bash, and through much trial and error I settled on the idiomatic manner. I'm not sure that we should deviate from it, as idiomatic Bash is our best bet at writing scripts many people can consume without having to think hard about the API.